io/include/pcl/io/pcd_io.h

```diff
@@ -252,12 +252,14 @@ namespace pcl
                             const Eigen::Quaternionf &orientation);
 
       /** \brief Generate the header of a BINARY_COMPRESSED PCD file format
+        * \param[out] os the stream into which to write the header
         * \param[in] cloud the point cloud data message
         * \param[in] origin the sensor acquisition origin
         * \param[in] orientation the sensor acquisition orientation
         */
-      std::string
-      generateHeaderBinaryCompressed (const pcl::PCLPointCloud2 &cloud,
+      int
+      generateHeaderBinaryCompressed (std::ostream &os,
```

Comment:
Agreed.  I apologize for that oversight - I think I regraded it as a private member, but I see it's public.